com.unity.render-pipelines.universal/ShaderLibrary/LightCookie/LightCookieInput.hlsl

```diff
@@ -75,43 +70,58 @@ int URP_LightCookie_GetLightType(int lightIndex)
     #endif
 }
 
-bool URP_LightCookie_MainLightTextureIsRGBFormat()
+bool IsMainLightCookieTextureRGBFormat()
 {
-    return _MainLightCookieFormat == URP_LIGHT_COOKIE_FORMAT_RGB;
+    return _MainLightCookieTextureFormat == URP_LIGHT_COOKIE_FORMAT_RGB;
 }
 
-bool URP_LightCookie_MainLightTextureIsAlphaFormat()
+bool IsMainLightCookieTextureAlphaFormat()
 {
-    return _MainLightCookieFormat == URP_LIGHT_COOKIE_FORMAT_ALPHA;
+    return _MainLightCookieTextureFormat == URP_LIGHT_COOKIE_FORMAT_ALPHA;
 }
 
-bool URP_LightCookie_AdditionalLightsTextureIsRGBFormat()
+bool IsAdditionalLightsCookieAtlasTextureRGBFormat()
 {
-    return _AdditionalLightsCookieAtlasFormat == URP_LIGHT_COOKIE_FORMAT_RGB;
+    return _AdditionalLightsCookieAtlasTextureFormat == URP_LIGHT_COOKIE_FORMAT_RGB;
 }
 
-bool URP_LightCookie_AdditionalLightsTextureIsAlphaFormat()
+bool IsAdditionalLightsCookieAtlasTextureAlphaFormat()
 {
-    return _AdditionalLightsCookieAtlasFormat == URP_LIGHT_COOKIE_FORMAT_ALPHA;
+    return _AdditionalLightsCookieAtlasTextureFormat == URP_LIGHT_COOKIE_FORMAT_ALPHA;
 }
 
 // Sampling
 
-real4 URP_LightCookie_SampleMainLightTexture(float2 uv)
+real4 SampleMainLightCookieTexture(float2 uv)
 {
     return SAMPLE_TEXTURE2D(_MainLightCookieTexture, sampler_MainLightCookieTexture, uv);
 }
 
-real4 URP_LightCookie_SampleAdditionalLightsTexture(float2 uv)
+real4 SampleAdditionalLightsCookieAtlasTexture(float2 uv)
 {
     return SAMPLE_TEXTURE2D(_AdditionalLightsCookieAtlasTexture, sampler_AdditionalLightsCookieAtlasTexture, uv);
 }
 
 // Helpers
 
-bool URP_LightCookie_IsEnabled(float4 uvRect)
+bool IsLightCookieEnabled(int lightBufferIndex)
 {
-    return all(uvRect == 0);
+#if 0
+    float4 uvRect = GetLightCookieAtlasUVRect(lightBufferIndex);
+    return any(uvRect != 0);
+#else
+    // 2^5 == 32, bit mask for a float/uint.
+    uint elemIndex = ((uint)lightBufferIndex) >> 5;
+    uint bitOffset = (uint)lightBufferIndex & ((1 << 5) - 1);
+
+    #if USE_STRUCTURED_BUFFER_FOR_LIGHT_DATA
+        uint elem = asuint(_AdditionalLightsCookieEnableBitsBuffer[elemIndex]);
+    #else
+        uint elem = asuint(_AdditionalLightsCookieEnableBits[elemIndex]);
+    #endif
+
+    return (elem & (1u << bitOffset)) != 0u;
+#endif
 }
```

Comment:
Will we enable the `#if 0 `section at some point?

Reply:
No, that's the old version. I left that in so I can do perf testing on mobile later (new on is on my macbook around 1% faster). Also it's an emergency switch to restore the old behavior, just in case. I'll remove it with my next PR.